pkg/storage/azure/azure.go

```diff
@@ -708,9 +773,83 @@ func (d *driver) assureStorageAccount(cfg *Azure, infra *configv1.Infrastructure
 		}
 	}
 
+	if !isAzureStackCloud(d.Config.CloudName) && cfg.FederatedTokenFile != "" {
+		azClient, err := d.newAzClient(cfg, environment, tagset)
+		if err != nil {
+			return "", false, err
+		}
+		err = azClient.DisableStorageAccountAccessKeyAccess(d.Context, cfg.ResourceGroup, accountName)
```

Comment:
@flavianmissi , does this part makes sense to you? This is for disabling the Access Key Access if the mode of authentication is FederatedTokenFile/Managed Identity.